WooCommerce/WooCommerceTests/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewModelTests.swift

```diff
@@ -0,0 +1,194 @@
+import XCTest
+import TestKit
+
+@testable import WooCommerce
+import Yosemite
+import Networking
+
+final class InPersonPaymentsMenuViewModelTests: XCTestCase {
+    private var stores: MockStoresManager!
+
+    private var storageManager: MockStorageManager!
+
+    private var noticePresenter: MockNoticePresenter!
+
+    private var analyticsProvider: MockAnalyticsProvider!
+    private var analytics: Analytics!
+
+    private var configuration: CardPresentPaymentsConfiguration!
+
+    private var dependencies: InPersonPaymentsMenuViewModel.Dependencies!
+
+    private var sut: InPersonPaymentsMenuViewModel!
+
+    private let sampleStoreID: Int64 = 12345
+
+    override func setUp() {
+        stores = MockStoresManager(sessionManager: .makeForTesting())
+        stores.sessionManager.setStoreId(sampleStoreID)
+        storageManager = MockStorageManager()
+        storageManager.insertSamplePaymentGateway(readOnlyGateway: PaymentGateway.fake().copy(siteID: sampleStoreID,
+                                                                                              gatewayID: "cod"))
+        noticePresenter = MockNoticePresenter()
+        analyticsProvider = MockAnalyticsProvider()
+        analytics = WooAnalytics(analyticsProvider: analyticsProvider)
+        configuration = CardPresentPaymentsConfiguration.init(country: "US")
+
+        dependencies = InPersonPaymentsMenuViewModel.Dependencies(
+            stores: stores,
+            storageManager: storageManager,
+            noticePresenter: noticePresenter,
+            analytics: analytics
+        )
+        sut = InPersonPaymentsMenuViewModel(dependencies: dependencies,
+                                            configuration: configuration)
+    }
+
+    // MARK: - Analytics tests
+    func test_updateCashOnDeliverySetting_enabled_tracks_paymentsHubCashOnDeliveryToggled_event() throws {
+        // Given
+
+        // When
+        sut.updateCashOnDeliverySetting(enabled: true)
+
+        // Then
+        assertNotEmpty(analyticsProvider.receivedEvents)
+        let indexOfEvent = try XCTUnwrap(analyticsProvider.receivedEvents.firstIndex(where: { $0 == AnalyticEvents.paymentsHubCashOnDeliveryToggled }))
+        let eventProperties = try XCTUnwrap(analyticsProvider.receivedProperties[indexOfEvent])
+        assertEqual("US", eventProperties[AnalyticProperties.countryCodeKey] as? String)
```

Comment:
As we use "US" across all tests, perhaps we could declare a dummy constant for these:

`private let sampleStoreCountryKey: String = "US"`

Reply:
We could, but I think it makes the test a bit more readable this way, and it's still a trivial change if we ever need to change the country in future. Not a big deal one way or the other, obviously, it's just a style preference.